datacube/drivers/postgis/_api.py

```diff
@@ -1474,33 +1500,40 @@ def remove_lineage_relations(self,
     def temporal_extent_by_prod(self, product_id: int) -> tuple[datetime.datetime, datetime.datetime]:
         query = self.temporal_extent_full().where(Dataset.product_ref == product_id)
         res = self._connection.execute(query)
-        return res.first()
+        for tmin, tmax in res:
+            return (self.time_min.normalise_value(tmin), self.time_max.normalise_value(tmax))
+        raise RuntimeError("Product has no datasets and therefore no temporal extent")
 
     def temporal_extent_by_ids(self, ids: Iterable[DSID]) -> tuple[datetime.datetime, datetime.datetime]:
         query = self.temporal_extent_full().where(Dataset.id.in_(ids))
         res = self._connection.execute(query)
-        return res.first()
+        for tmin, tmax in res:
+            return (self.time_min.normalise_value(tmin), self.time_max.normalise_value(tmax))
+        raise ValueError("no dataset ids provided")
+
+    time_min = DateDocField('acquisition_time_min',
```

Comment:
They are very specialised for this API method, and because they contain aggregation would break the query if you tried to use them in a different context.  So I'd be reluctant to put them in `_dataset_fields`.

Reply:
Sorry, a poorly-worded comment - I meant literally move it further towards the top of the code alongside our other 'constant', `_dataset_fields`, rather than sandwiched between two methods